var/spack/repos/builtin/packages/catch/package.py

```diff
@@ -63,6 +63,9 @@ def build(self, spec, prefix):
     @when('+single_header')
```

Comment:
@ax3l I didn't examine the rest of the file closely when I reviewed this and only checked the logic related to installation of the headers.

I think this package could be improved by:

* removing the `single_header` variant: this doesn't need to be decided by a variant. The version decides whether a single header should be installed.
* Likewise the conditionally-defined `cmake`/`build` methods should be conditioned on versions rather than the variant

Reply:
Can you elaborate how the version can be the `single_header` discriminator? Potentially, a single header install can differ from a "proper" install (e.g. CMake config files, etc.) which makes this a candidate for a variant for me.

Reply:
I think I misunderstood now that I look at this again.

If I understand now: this package has always had an option to install normally or as a single header. As of 2.3.0, the single header install method changed. Is that right?

Reply:
Close: Independent of the variant of install, the public headers have changed. They did their breaking API change for include paths a bit "late" for Catch2 ;-)